File: backend/src/main/java/com/culinaryheaven/domain/bookmark/dto/response/BookMarksResponse.java
package com.culinaryheaven.domain.bookmark.dto.response;

import com.culinaryheaven.domain.bookmark.domain.BookMark;
import com.culinaryheaven.domain.recipe.dto.response.RecipeResponse;
import com.culinaryheaven.domain.recipe.dto.response.RecipesResponse;
import org.springframework.data.domain.Page;

import java.util.List;

public record BookMarksResponse(
        List<BookMarkResponse> bookMarks
) {

    public static BookMarksResponse of(Page<BookMark> bookMarks) {
        List<BookMarkResponse> bookMarksResponse = bookMarks.getContent()
                .stream()
                .map(BookMarkResponse::of)
                .toList();

        return new BookMarksResponse(bookMarksResponse);

    }
}

File: android/data/src/main/java/com/jae464/data/remote/model/response/bookmark/BookMarksResponse.kt
package com.jae464.data.remote.model.response.bookmark

import com.jae464.domain.model.BookMark
import kotlinx.serialization.Serializable

@Serializable
data class BookMarksResponse(
    val bookMarks: List<BookMarkResponse>
)

fun BookMarksResponse.toDomain(): List<BookMark>  {
    return bookMarks.map { it.toDomain() }
}